### DISCUSSION TITLE
Allow for setting multiple API keys in provider (etherscan.io, Alchemy, infura, etc.)

### DISCUSSION THREAD

**Discussion by o-az** [wevm/wagmi#67] [Ideas]
> ( TL; DR – sign up for your own API keys with the links below to improve your application performance )

It looks like we can only do InfuraId for specific connectors.

With the current setup it's bound to cause some throttle. According to ethers.js, it's advisable to add API keys. Example:

```JavaScript
// Specify your own API keys
// Each is optional, and if you omit it the default
// API key for that service will be used.
const provider = ethers.getDefaultProvider("homestead", {
    etherscan: YOUR_ETHERSCAN_API_KEY,
    infura: YOUR_INFURA_PROJECT_ID,
    // Or if using a project secret:
    // infura: {
    //   projectId: YOUR_INFURA_PROJECT_ID,
    //   projectSecret: YOUR_INFURA_PROJECT_SECRET,
    // },
    alchemy: YOUR_ALCHEMY_API_KEY,
    pocket: YOUR_POCKET_APPLICATION_KEY
    // Or if using an application secret key:
    // pocket: {
    //   applicationId: ,
    //   applicationSecretKey:
    // }
});
```
https://docs.ethers.io/v5/api-keys/

The following message is logged in the browser if I don't provide API keys:

> ====== NOTICE ======:
>Request-Rate Exceeded (this message will not be repeated)
>
>The default API keys for each service are provided as a highly-throttled,
community resource for low-traffic projects and early prototyping.
>
>While your application will continue to function, we highly recommended
signing up for your own API keys to improve performance, increase your
request rate/limit and enable other perks, such as metrics and advanced APIs.
>
>For more details: https://docs.ethers.io/api-keys/

**Comment by tmm** [comment]
[Added a note](https://wagmi-xyz.vercel.app/docs/provider#provider-optional) to the docs about settings API keys for `getDefaultProvider` to minimize disruption.

The WalletConnect and WalletLink connectors support Infura, Alchemy, and other RPC service providers. For example, using the `WalletConnectConnector`:

```ts
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'

const infuraWalletConnectConnector = new WalletConnectConnector({
  options: {
    infuraId: 'Your infura id',
  },
})

// Little less intuitive, but it still works!
// This is just how WalletConnect's Ethereum Provider options are set up.
const alchemyWalletConnectConnector = new WalletConnectConnector({
  options: {
    rpc: 'https://eth-mainnet.alchemyapi.io/v2/yourAlchemyId',
  },
})
```

**Comment by ssuvorov** [comment]
Is there a way to pass a etherscan API token? e.g.
```
getDefaultProvider('goerli', {
      etherscan: "ETHERSCAN_API_TOKEN"
    })
```
